### ISSUE TITLE
Reduce PDF file size by completely discarding the clipped out part of an image

### ISSUE THREAD

**Issue by Andrew15-5** [typst/typst#5213]
### Description

Imagine you have an image that you want to have only a part of. You would use something like `block(clip: true, image("...")` and make sure that some part of the image you don't want isn't visible (https://github.com/typst/typst/issues/2709#issuecomment-2411709983). But now, even if you only see a part of the image, you probably would like the image to be actually cropped, meaning that the invisible part should be removed permanently (from the PDF file).

I think this can either be a field like `block(clip: true, crop: true)` which is turned on by default, or a dedicated function like `crop()`.

<details><summary>Tagged PDF</summary>

There were also some concerns about that, but I have no idea what this means exactly, but still want to mention that.

</details>

A small conversation started [here](https://discord.com/channels/1054443721975922748/1088371867913572452/1295789520213643274).

### Use Case

The smaller the PDF file size, the better!

**Comment by TS53** [comment]
Unfortunately, none of the supported image formats implement [ImageDecoderRect](https://docs.rs/image/latest/image/trait.ImageDecoderRect.html). I think this would not make the compiler faster and probably even slower.
If this gets implemented, the compiler should check if the cropped images filesize is actually smaller before including the cropped version. Otherwise stay with the default method.

If I remember correctly, JPG or a version of JPG could get embedded in the PDF without reencoding, so basically just copying the file into the binary without opening it. If this is true, the compiler will definetly become slower when including a jpg image and cropping it.

**Comment by reknih** [comment]
> If I remember correctly, JPG or a version of JPG could get embedded in the PDF without reencoding, so basically just copying the file into the binary without decoding it.

This is true, but only with the caveat that the crop has to align with the JPEG chunks (most of the time 8x8). The way would be to crop the JPEG to the nearest chunk boundary outside of the clipping area and then recalculate its position. This should shrink the PDF at no loss of quality and that operation would not be too expensive. `jpegtran` does it. Not aware of a feature in `zune-jpeg` that does that, though.
